random_question/weighted_random.py

```diff
@@ -10,11 +10,26 @@
 
 # TODO: Hess should implement!
 def weighted_sampler(data, weights):
-    """Should return a random item from data array, weighted by the weights array"""
-    raise NotImplementedError("Hess should implement!")
+    random_number=random.random()
+    print('this is the:')
+    print(random_number)
+    Dfunction = arrange_array(weights)
+    print(Dfunction)
+    for x in range(len(Dfunction)):
```

Comment:
no need to iterate on a range. in python you can iterate directly on items of any iterable:

`for num in Dfunction: ...`

Reply:
אבל אי צריך גם את האיבר הבא ! :( איך עושים את זה בצורה חכמה ?

Reply:
if you have a better way to write it i will be glad to see :)